modules/gapi/src/backends/common/serialization.hpp

```diff
@@ -256,6 +243,70 @@ I::IStream& operator>> (I::IStream& is, cv::util::variant<Ts...> &v) {
     return detail::get_v<cv::util::variant<Ts...>, Ts...>(is, v, 0u, idx);
 }
 
+// Generic: vector serialization ///////////////////////////////////////////////
+// Moved here to fix CLang issues https://clang.llvm.org/compatibility.html
+// Unqualified lookup in templates
+template<typename T>
+I::OStream& operator<< (I::OStream& os, const std::vector<T> &ts) {
+    //const std::size_t sz = ts.size(); // explicitly specify type
+    const uint32_t sz = (uint32_t)ts.size(); // explicitly specify type
+    os << sz;
+    for (auto &&v : ts) os << v;
+    return os;
+}
+template<typename T>
+I::IStream& operator >> (I::IStream& is, std::vector<T> &ts) {
+    //std::size_t sz = 0u;
+    uint32_t sz = 0u;
+    is >> sz;
+    if (sz == 0u) {
+        ts.clear();
+    }
+    else {
+        ts.resize(sz);
+        for (auto &&i : ade::util::iota(sz)) is >> ts[i];
+    }
+    return is;
+}
+
+inline I::OStream& operator<< (I::OStream& os, const cv::GRunArg &arg) {
+    os << (uint32_t)arg.index();
+    switch (arg.index()) {
+#if !defined(GAPI_STANDALONE)
+    case cv::GRunArg::index_of<cv::UMat>() :
+        os << cv::util::get<cv::UMat>(arg);       break;
+#endif
+    case cv::GRunArg::index_of<cv::Mat>() :
+        os << cv::util::get<cv::Mat>(arg);        break;
+    case cv::GRunArg::index_of<cv::Scalar>() :
+        os << cv::util::get<cv::Scalar>(arg);     break;
+    default: GAPI_Assert(false && "GRunArg: Unsupported type for <<");
+    }
+    return os;
+}
+inline I::IStream& operator >> (I::IStream& is, cv::GRunArg &arg) {
+    uint32_t index;
+    is >> index;
+#if !defined(GAPI_STANDALONE)
+    cv::UMat  v0;
+#endif
+    cv::Mat v1;
+    cv::Scalar v2;
+    switch (index) {
+#if !defined(GAPI_STANDALONE)
+    case cv::GRunArg::index_of<cv::UMat>() :
+        is >> v0; arg = v0; break;
+#endif
+    case cv::GRunArg::index_of<cv::Mat>() :
+        is >> v1; arg = v1; break;
+    case cv::GRunArg::index_of<cv::Scalar>() :
+        is >> v2; arg = v2; break;
+    default: GAPI_Assert(false && "GRunArg: Unsupported type for >>");
+    }
+    return is;
+}
```

Comment:
we don't need this method here, overload via `<< (variant<T...>)` should work instead

Reply:
Done